get_deps.sh

```diff
@@ -1,142 +1,185 @@
 #!/usr/bin/env bash
 
-set -e
-set -x
-
-BASE_DIRECTORY=`pwd`
+HERE="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
 
-# Allow different deps for different platforms:
-if [ -z "$DEPS_DIRECTORY" ]; then
-  DEPS_DIRECTORY=${BASE_DIRECTORY}/deps
+set -e
+[[ $VERBOSE == 1 ]] && set -x
+
+if [[ "$1" == "cpu" ]]; then
+	GPU=no
+	VARIANT=cpu
+elif [[ "$1" == "gpu" ]]; then
+	GPU=yes
+	VARIANT=gpu
+else
+	GPU=${GPU:-no}
+	if [[ $GPU == 1 ]]; then
+		VARIANT=gpu
+	else
+		VARIANT=cpu
+	fi
 fi
 
-mkdir -p ${DEPS_DIRECTORY}
-cd ${DEPS_DIRECTORY}
-DEPS_DIRECTORY=$PWD # -- to avoid relative/absolute confusion
+DEPS_DIR=$HERE/deps
+mkdir -p ${DEPS_DIR}
+cd ${DEPS_DIR}
 
-PREFIX=${DEPS_DIRECTORY}/install
+PREFIX=${DEPS_DIR}/install-$VARIANT
```

Comment:
I'm not convinced having a `deps/install-cpu` and `deps/install-gpu` is the right thing to do.

If anything, we should keep the deps separated by backend, like

```
deps/install/libtensorflow-cpu/
deps/install/libtensorflow-gpu/
deps/install/libtorch-cpu/
deps/install/libtorch-gpu/
...
```

or actually, with an explicit version

```
deps/install/libtensorflow-1.12.0-cpu/
deps/install/libtensorflow-1.12.0-gpu/
deps/install/libtorch-1.2-cpu/
deps/install/libtorch-1.2-gpu/
...
```

so that it's easier to know where they are and to link to them from the backends.

Reply:
Since the device spec (CPU=1/GPU=1) is dominant, it looks easier to collect all dependencies under an encompassing directory. But then again, I'd like to first move into the bin/arch-os-variant form, and continue from there.

Reply:
What I mean is that it is dominant for each individual backend, not for RedisAI as a whole.
It's really important that we get this right, I don't think we are 100% aligned here (see my previous comment on bin/arch-os-variant)